src/templates/finops-hub/modules/hub.bicep

```diff
@@ -83,6 +87,70 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2022-09-01' = if (ena
   }
 }
 
+//------------------------------------------------------------------------------
+// Register EventGrid provider so that the data factory trigger can run.
+//------------------------------------------------------------------------------
+
+resource RegisterEventGridProvider 'Microsoft.EventGrid/namespaces@2023-12-15-preview' = {
+  name: '${uniqueSuffix}-registerRP'
+  location: location
+  sku: {
+    capacity: 1
+    name: 'Standard'
+  }
+  properties: {
+    publicNetworkAccess: 'Enabled'
```

Comment:
Given some orgs may have a policy to disable public network access, should we set this to disabled or whatever the most restrictive value is?

Reply:
Could we put the register provider PS commands into the script we use to start the triggers after deployment rather than deploying a dummy event grid and deleting it afterwards? We need to register the provider before starting the trigger in ADF.  It's not a dependency for deploying the trigger, just starting it.

Reply:
The goal here is to avoid the registration step because registration requires subscription-level access. If we deploy an EventGrid resource, then ARM will do the registration automatically.

Reply:
done